pkg/agent/config/node_config.go

```diff
@@ -268,12 +268,18 @@ func (nc *NetworkConfig) CalculateMTUDeduction(isIPv6 bool) int {
 	var mtuDeduction int
 	// When Multi-cluster Gateway is enabled, we need to reduce MTU for potential cross-cluster traffic.
 	if nc.TrafficEncapMode.SupportsEncap() || nc.EnableMulticlusterGW {
-		if nc.TunnelType == ovsconfig.VXLANTunnel {
-			mtuDeduction = vxlanOverhead
-		} else if nc.TunnelType == ovsconfig.GeneveTunnel {
-			mtuDeduction = geneveOverhead
-		} else if nc.TunnelType == ovsconfig.GRETunnel {
-			mtuDeduction = greOverhead
+		if nc.TrafficEncryptionMode == TrafficEncryptionModeWireGuard {
+			mtuDeduction = WireGuardOverhead
+		} else if nc.TrafficEncryptionMode == TrafficEncryptionModeIPSec {
+			mtuDeduction = IPSecESPOverhead
```

Comment:
Thanks for the comment, I removed the return value of `CalculateMTUDeduction`, when we need to calculate MTUs, the caller just uses `networkConfig.MTUDeduction` directly. And when the `TrafficEncryptionMode` is WireGuard, it will deduct the WG overhead.